src/main/java/seedu/uninurse/model/task/TaskList.java

```diff
@@ -118,6 +118,30 @@ public boolean containsTaskToday() {
         return !this.getAllTasksToday().isEmpty();
     }
 
+    /**
+     * Updates the internaltasklist with all the new tasks generated because of RecurringTasks
+     * and their next RecurringTask once past the task date.
+     */
+    public void updateRecurringTasks() {
+        ArrayList<Task> updatedTasks = new ArrayList<>(internalTaskList);
+
+        for (Task task : internalTaskList) {
+            if (task instanceof RecurringTask && ((RecurringTask) task).pastTaskDate()) {
+                RecurringTask recurringTaskTracker = (RecurringTask) task;
+                // This is to populate all the tasks up till this point in case the user inputs a past date for the task
+                while (recurringTaskTracker.pastTaskDate()) {
+                    RecurringTask nextRecurringTask = recurringTaskTracker.getNextRecurringTask();
+                    if (!updatedTasks.contains(nextRecurringTask)) {
+                        updatedTasks.add(nextRecurringTask);
+                    }
+                    recurringTaskTracker = nextRecurringTask;
+                }
+            }
+        }
```

Comment:
Although this code segment works, I would suggest not use typecasting as Java forcefully typecasts objects and would be bad in general if used outside of overriding `equals()` methods.

I would suggest using a `RecurrableTask` interface (although not the best name since `Task` is not recurrable) that provides the methods `RecurrableTask#isRecurring()` and `RecuccableTask#getNextRecurringTask()`.
Then both `Task` and `RecurringTask` would both implement these 2 methods provided by the interface, where `Task` will simply return false and return an `Optional<RecurrableTask>` that is an `Optional.empty` for those 2 methods respectively.

Hence this code segment can be simplified to
```java
for (RecurrableTask task : internalTaskList) {
            if (task.isRecurring()) {
                RecurrableTask recurringTaskTracker = task;
                // This is to populate all the tasks up till this point in case the user inputs a past date for the task
                while (recurringTaskTracker.pastTaskDate()) {
                    RecurringTask nextRecurringTask = recurringTaskTracker.getNextRecurringTask()
                            .orElseThrow(() -> new NoRecurringTaskException());
                    if (!updatedTasks.contains(nextRecurringTask)) {
                        updatedTasks.add(nextRecurringTask);
                    }
                    recurringTaskTracker = nextRecurringTask;
                }
            }
        }
```

Reply:
Actually after some thought I came up with a cleaner solution that does not require you to add an interface.
Since recurring tasks are already dynamically added when it is past the current date, you should just overload the method `TaskList#add()` to accept `RecurringTask` and hence you would not require an `updateRecurringTask()` in Model as well since it would not have a use outside of adding a recurring task.

Hence you can ignore the previous suggestion of adding an interface and using `Optional<T>`s

```java
public TaskList add(RecurringTask recurringTask) {
        List<Task> updatedTasks = new ArrayList<>(internalTaskList);
        updatedTasks.add(recurringTask);
        RecurrableTask recurringTaskTracker = recurringTask;
                // This is to populate all the tasks up till this point in case the user inputs a past date for the task
                while (recurringTaskTracker.pastTaskDate()) {
                    RecurringTask nextRecurringTask = recurringTaskTracker.getNextRecurringTask();
                    if (!updatedTasks.contains(nextRecurringTask)) {
                        updatedTasks.add(nextRecurringTask);
                    }
                    recurringTaskTracker = nextRecurringTask;
                }
        return new TaskList(updatedTasks);
    }
```

Reply:
Hey, recurring tasks are added dynamically because of the `updateRecurringTask()` right. Otherwise they only time the model will get updated is when I add another RecurringTask or I restart the application (although this behavior makes perfect sense, I wanted to be such that even if the user keeps the application on forever, it would automatically update the recurring tasks whenever the user executes any command)

Reply:
That is a valid point. I think we can refine the behavior of it in the next iteration.